main.go

```diff
@@ -174,10 +177,25 @@ func main() {
 		os.Exit(1)
 	}
 
+	if len(*argTopicBasePath) > 0 {
+		topicNameTemplate = strings.Replace(topicNameTemplate, "internal/mqtt-stresser", *argTopicBasePath, 1)
+	}
+
 	payloadGenerator := defaultPayloadGen()
 	if len(*argConstantPayload) > 0 {
-		verboseLogger.Printf("Set constant payload to %s\n", *argConstantPayload)
-		payloadGenerator = constantPayloadGenerator(*argConstantPayload)
+		if strings.Index(*argConstantPayload, "@") == 0 {
```

Comment:
I think we should move this logic to the constantPayloadGenerator function or build a new function around it to keep things were they belong.

Reply:
good idea.